src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs

```diff
@@ -327,11 +315,31 @@ private void CreateSimpleServicePrincipal()
                 }
             };
 
-            var shouldProcessMessage = SkipRoleAssignment() ?
-                                        string.Format("Adding a new service principal to be associated with an application " +
-                                                      "having AppId '{0}' with no permissions.", createParameters.ApplicationId) :
-                                        string.Format("Adding a new service principal to be associated with an application " +
+            var shouldProcessMessage = string.Format("Adding a new service principal to be associated with an application " +
+                                                      "having AppId '{0}' with no permissions.", createParameters.ApplicationId);
+
+            if (!SkipRoleAssignment())
+            {
+                var subscriptionId = DefaultContext.Subscription.Id;
+                if (!this.IsParameterBound(c => c.Scope))
+                {
+                    Scope = string.Format("/subscriptions/{0}", subscriptionId);
+                    WriteVerbose(string.Format("No scope provided - using the default scope '{0}'", Scope));
+                }
+
+                AuthorizationClient.ValidateScope(Scope, true);
+
+                if (!this.IsParameterBound(c => c.Role))
+                {
+                    Role = "Contributor";
+                    WriteVerbose(string.Format("No role provided - using the default role '{0}'", Role));
+                }
+                WriteWarning(string.Format("Assigning role '{0}' over scope '{1}' to the new service principal.", this.Role, this.Scope));
```

Comment:
I suppose this warning message should still be in the `if (shouldprocess())` block, because we don't actually assign role until user confirmed 🤔